src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,109 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int CAPACITY_MULTIPLIER = 2;
+    private int size = 0;
+    private Node<K, V>[] hashTable = new Node[INITIAL_CAPACITY];
+    private int threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    private int currentCapacity = INITIAL_CAPACITY;
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> ifNodeExistForKey = returnNodeOfKey(key);
+        if (ifNodeExistForKey != null) {
+            ifNodeExistForKey.value = value;
+            return;
+        }
+        putWithoutCheckKeyContaince(key, value);
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> ifNodeExistForKey = returnNodeOfKey(key);
+        if (ifNodeExistForKey != null) {
+            return ifNodeExistForKey.value;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
+        K key;
+        V value;
+        Node<K, V> prev;
```

Comment:
Having double linked list instead of single linked list doesn't give us benefit but makes code more complicated comparing to single linked list. In future try to stick to KISS principle. For now you don't need to rewrite this solution just fix comments if left and I'll approve